Sources/General/Commands/Setup.swift

```diff
@@ -0,0 +1,337 @@
+        let destination = getTemplatesDestination()
+
+        var moved = [URL]()
+        var isSpecModified = false
+        do {
+            moved = try move(setupFiles.templates, to: destination)
+            if let url = setupFiles.spec {
+                isSpecModified = updateSpec(templateURL: url)
+            }
+        }
+        catch {
+            try remove(folderURL)
+            throw error
+        }
+        try remove(folderURL)
+        print()
+        displayTemplatesResult(moved, isSpecModified: isSpecModified)
+    }
+
+    private func getGitRepoPath() throws -> String {
+        let components = githubPath.split(separator: " ")
+        guard let name =  components.first else {
+            throw Error.githubName(githubPath)
+        }
+        var branch = "master"
+        if components.count > 1 {
+            branch = String(components[1])
+        }
+
+        return "https://github.com/\(name)/archive/\(branch).zip"
+    }
+
+    private func downloadArchive(at path: String) throws -> URL {
+        guard let url = URL(string: path) else {
+            throw Error.url(path)
+        }
+        let data = try self.data(at: url)
+        let tmpFolderURL = try tempFolderWithRandomName()
+        let fileURL = tmpFolderURL + url.lastPathComponent
+        do {
+            try data.write(to: fileURL)
+        }
+        catch {
+            throw Error.write(tmpFolderURL)
+        }
+        return fileURL
+    }
+
+    private func data(at url: URL) throws -> Data {
+        do {
+            return try Data(contentsOf: url)
+        }
+        catch {
+            throw Error.download(url)
+        }
+    }
+
+    private func tempFolderWithRandomName() throws -> URL {
+        let root = fileManager.homeDirectoryForCurrentUser
+        let url = root + Constants.appFolderName + "tmp" + UUID().uuidString
+        do {
+            try fileManager.createDirectory(at: url, withIntermediateDirectories: true, attributes: nil)
+        }
+        catch {
+            throw Error.write(url)
+        }
+        return url
+    }
+
+    private func unzipArchive(at url: URL) throws -> URL {
+        let folderURL = url.deletingLastPathComponent()
+        do {
+            try fileManager.unzipItem(at: url, to: folderURL)
+        }
+        catch {
+            throw Error.write(folderURL)
+        }
+        return folderURL
+    }
+
+    private func loadSteupFiles(in url: URL) -> (templates: [URL], spec: URL?) {
+        var templates = [URL]()
+        var spec: URL?
+        do {
+            let contents = try fileManager.contentsOfDirectory(at: url,
+                                                               includingPropertiesForKeys: nil,
+                                                               options: [])
+            for url in contents {
+                var isDirectory: ObjCBool = false
+                _ = fileManager.fileExists(atPath: url.path, isDirectory: &isDirectory)
```

Comment:
You should be able to check for directory using `isDirectoryKey` for `includingPropertiesForKeys` and checking for it's resource value without needing an additional `fileExists` call.